src/main/scala/strawman/collection/Set.scala

```diff
@@ -42,9 +58,13 @@ trait SetLike[A, +C[X] <: Set[X]]
 trait SetMonoTransforms[A, +Repr]
   extends IterableMonoTransforms[A, Repr] {
 
-  def & (that: Set[A]): Repr
+  def & (that: Set[A]): Repr = this.filter(that)
+
+}
+
+trait SetPolyTransforms[A, +C[X]] extends IterablePolyTransforms[A, C] {
 
-  def ++ (that: Set[A]): Repr
+  def ++ (that: Set[A]): C[A]
```

Comment:
One good reason to keep immutable Set invariant is that many immutable sets are based on comparators and it's not that easy to write a compactor that is able to compare all the children of a non-sealed class. But you want to be able to put  classes from non-sealed higherarchies into immutable sets.